R-package/R/lgb.Booster.R

```diff
@@ -149,7 +149,7 @@ Booster <- R6::R6Class(
     add_valid = function(data, name) {
 
       # Check if data is lgb.Dataset
-      if (!lgb.check.r6.class(data, "lgb.Dataset")) {
+      if (!(lgb.Dataset = lgb.check.r6.class(data, "lgb.Dataset"))) {
```

Comment:
this check is still incorrect. Please accept this suggestion in your browser.

```suggestion
      if (!lgb.check.r6.class(object = data, name = "lgb.Dataset")) {
```